usaspending_api/disaster/v2/views/agency/loans.py

```diff
@@ -44,25 +43,49 @@ def route_agency_loans_backend(request, *args, **kwargs):
     return route_agency_loans_backend
 
 
-class LoansByAgencyViewSet(LoansPaginationMixin, LoansMixin, FabaOutlayMixin, DisasterBase):
+class LoansByAgencyViewSet(
+    LoansPaginationMixin, ElasticsearchAccountDisasterBase, LoansMixin, FabaOutlayMixin, DisasterBase
+):
     """
         This endpoint provides insights on the Agencies awarding loans from
         disaster/emergency funding per the requested filters.
     """
 
     endpoint_doc = "usaspending_api/api_contracts/contracts/v2/disaster/agency/loans.md"
+    required_filters = ["def_codes", "query"]
+    query_fields = ["funding_toptier_agency_name.contains"]
+    agg_key = "financial_accounts_by_award.funding_toptier_agency_id"  # primary (tier-1) aggregation key
+    nested_nonzero_fields = {"outlay": "gross_outlay_amount_by_award_cpe", "obligation": "transaction_obligated_amount"}
+    top_hits_fields = [
+        "financial_accounts_by_award.funding_toptier_agency_code",
+        "financial_accounts_by_award.funding_toptier_agency_name",
+    ]
 
     @cache_response()
     def post(self, request):
+        self.filters.update({"award_type": ["07", "08"]})
```

Comment:
The `loans` endpoints don't actually allow for the `award_type_codes` filter, they get removed by tinyshield. If we want to allow API users to specify which loan types they want, it's an easy change, just not what the endpoint is doing currently

Reply:
I'm okay with doing whatever the api contract says it should do. If the api contract says you should be able to provide arbitrary loan types, then we need to support that @emily-brents

Reply:
Looking at the api docs, this is a bit more complicated. 

Federal Account and object class show using it in a sample request body, but don't document it as an available filter. Probably we should just take it out of the sample request body if the API request works without it.

Agency however targets 2 different agency sums depending on whether any award type codes, or NO award type codes filters is provided. So it's important how the presence of absence of `award_type_codes` is handled in this particular endpoint, as it should route to different backend queries. **Is that still the case after your implementation; i.e. does it do what is documented in the API docs for [`agency/loans.md`](https://github.com/fedspendingtransparency/usaspending-api/blob/master/usaspending_api/api_contracts/contracts/v2/disaster/agency/loans.md)?**

- https://github.com/fedspendingtransparency/usaspending-api/blob/master/usaspending_api/api_contracts/contracts/v2/disaster/object_class/loans.md
- https://github.com/fedspendingtransparency/usaspending-api/blob/master/usaspending_api/api_contracts/contracts/v2/disaster/federal_account/loans.md
- https://github.com/fedspendingtransparency/usaspending-api/blob/master/usaspending_api/api_contracts/contracts/v2/disaster/agency/loans.md

Reply:
I thought `award_type_codes` were expected to enable different behavior:

```+ `award_type_codes` (optional, array[enum[string]], fixed-type)
    Only accepts loan award type `07` or `08` in the array, since this endpoint is specific to loans.

    If ANY award type codes are provided, loan amounts will be summed for the distinct set of toptier agencies,
    whose subtier agencies funded loan awards linked to `FinancialAccountsByAwards` records (which are derived from DABS File C).

    If this parameter is not provided, loan amounts will be summed for a different set of agencies:
    the distinct set of toptier agencies "owning" appropriations accounts used in funding _any_ award spending
    for this disaster (i.e. from agencies "owning" Treasury Account Symbol (TAS) accounts on `FinancialAccountsByAwards`
    records, which are derived from DABS File C).
```

Reply:
The agency loans endpoint still works as expected - i.e., you can provide either "07" or "08" and the results will be filtered by the appropriate award type and the subtier agencies will be displayed.